docs/content/preview/releases/techadvisories/ta-26666.md

```diff
@@ -0,0 +1,92 @@
+
+BEGIN;
+INSERT INTO child_table(fkey) VALUES(..);
+COMMIT;
+```
+
+## Background
+
+The transaction inserts the row to the child table, and the insert generates intents on the child table's tablet. As this insert references the parent tablets row, a lock record is generated corresponding to the row on the parent tablet's row (in order to maintain the foreign key referential integrity). After the transaction commits, these intents can be applied to regular RocksDB and no longer be maintained in the system. In every tablet (parent and child), DocDB apply process adds the transaction that generates intents to a map. This map is used to determine the set of transactions that the transaction loader loads during the tablet bootstrap process and is necessary to ensure that all the transactions necessary for bootstrap are loaded. This map (consisting of transaction IDs), is persisted along with retryable requests structure at the time of transaction log (WAL) rollover, that is used by tablet bootstrap.
+
+## Bug
+
+The map that tracks the transactions grows unbounded on the parent tablet due to the following reasons:
+
+- The current logic to trim the map occurs during transaction log (WAL) rollover for that tablet, so unless there is enough WAL activity on the tablet, this map ends up consuming memory (even though those transactions have committed from a user perspective, applied from the DocDB perspective, and are no longer necessary for bootstrap).
+
+- If the workload only generates a lot of write traffic to the child table, then the intents generated for the parent table's tablets do not have any write_batches, as a result the existing trimming logic that relies on checking `max_replicated_op_id` doesn't qualify as `max_replicated_op_id` does not take effect, rendering the existing cleanup logic useless causing the memory of the map to grow unbounded.
+
+Currently, the memory corresponding to this map is untracked. If the memory growth is proportional to the number of tablets on the node (can be unbounded in some cases), it can lead to TServer soft memory limit being hit and workload would get throttled.
+
+### Other impacted scenarios
+
+Workloads that use serializable isolation and have read only traffic suffer from similar problems. This is less common compared to the described foreign key workloads.
+
+## Identifying the issue
+
+The metric `wal_replayable_applied_transactions` can be used to determine the current size of the map in question.
+
+For example, the following PromQL query shows the approximate memory overhead of the map in MB. (Each entry is about 64 bytes. So you multiply by 64 and divide by 1024*1024 to get the number of MBs.)
+
+```promql
+wal_replayable_applied_transactions{export_type="tserver_export", node_prefix="yb-dev-rthallam-csgyj"}*64/(1024*1024)
+```
+
+Heap snapshot URL on the TServer, would show a lot of allocations from the `TransactionParticipant::Impl::RemoveTransaction()` as per the following illustration:
```

Comment:
What is the Heap snapshot URL on the TServer??

Reply:
cc @rthallamko3